runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -400,6 +418,8 @@ public Instant get() {
      *     <ul>
      *       <li>the current input watermark</li>
      *       <li>the current watermark holds</li>
+     *       <li>the earliest delivery timestamp of pending timers</li>
```

Comment:
I believe now that you have output timestamp, you don't need to take the minimum of the delivery timestamp. That is how you can use this for GC / OnWindowExpiration.